util/db_utils.go

```diff
@@ -0,0 +1,132 @@
+package util
+
+import (
+	"database/sql"
+	"fmt"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/sirupsen/logrus"
+	"github.com/test-network-function/collector/types"
+)
+
+func HandleTransactionRollback(tx *sql.Tx, context string, err error) {
+	txErr := tx.Rollback()
+	if txErr != nil {
+		logrus.Errorf(RollbackErr, txErr)
+	}
+	logrus.Errorf(context, err)
+}
+
+func GetEntireCollectorTable(db *sql.DB) (claimRows, claimResultsRows *sql.Rows) {
+	claimRows, err := db.Query(SelectAllFromClaim)
+	if err != nil {
+		logrus.Errorf(ExecQueryErr, err)
+	}
+
+	claimResultsRows, err = db.Query(SelectAllFromClaimResult)
+	if err != nil {
+		logrus.Errorf(ExecQueryErr, err)
+	}
+
+	return claimRows, claimResultsRows
+}
+
+func GetCollectorTablesByPartner(db *sql.DB, partnerName string) (claimRows, claimResultsRows *sql.Rows) {
+	claimRows, err := db.Query(SelectAllFromClaimByPartner, partnerName)
+	if err != nil {
+		logrus.Errorf(ExecQueryErr, err)
+	}
+
+	// Extract claim IDs of given partner
+	claimIDsRows, err := db.Query(SelectAllClaimIDsByPartner, partnerName)
+	if err != nil {
+		logrus.Errorf(ExecQueryErr, err)
+	}
+	defer claimIDsRows.Close()
+
+	var claimIDsList []string
+	for claimIDsRows.Next() {
+		var claimID string
+		claimIDErr := claimIDsRows.Scan(&claimID)
+		if err != nil {
+			logrus.Errorf(ScanDBFieldErr, claimIDErr)
+		}
+		claimIDsList = append(claimIDsList, claimID)
+	}
+
+	// Extract claim results of found claim IDs
+	claimResultsQuery := fmt.Sprintf(SelectAllFromClaimResultByClaimIDs, strings.Join(claimIDsList, ","))
+	claimResultsRows, err = db.Query(claimResultsQuery)
+	if err != nil {
+		logrus.Errorf(ExecQueryErr, err)
+	}
+
+	return claimRows, claimResultsRows
+}
+
+// This function stores the claim and claim result into the database in a transaction
+func StoreClaimFileInDatabase(w http.ResponseWriter, r *http.Request, db *sql.DB, claimFileMap map[string]interface{}, claimResult []types.ClaimResult, partnerName, executedBy, ocpVersion string) bool {
```

Comment:
The following arguments are not used: w http.ResponseWriter, r *http.Request, claimFileMap map[string]interface{}

Reply:
Right, I forgot to remove these.